.github/scripts/branch_name_validation.sh

```diff
@@ -7,7 +7,7 @@ local_branch=${1}
 valid_branch="^[a-z][a-z-0-9-]*$"
 
 
-if [[ ! $local_branch =~ $valid_branch ]] && [[ $local_branch -gt 128 ]]; then
+if [[ ! $local_branch =~ $valid_branch ]] || [[ $local_branch -gt 128 ]]; then
```

Comment:
Maybe the regex should be `"^[a-z][a-z0-9-]*$"` 
Note the hyphen missing after the `z`.
For example, a567b is a match for `"^[a-z][a-z0-9-]*$"`  but it doesn't match the original regex with the extra hyphen.

Reply:
Nice catch @leslie-corbalt!  The second check for a long branch name is also not working, I'll fix that as well.

Reply:
Thinking about this more:  the full service name has a constraint of 128 characters, because that's what is imposed by [CloudFormation for stack names](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/cfn-using-console-create-stack-parameters.html).  So the actual constraint on the branch/stage name is less than 128, because the formula used for stack names is {service}-{branch/stage}.  The actual formula is `128 - {longest service name} - {one hyphen}`

Reply:
I'm going to scope this to just fixing the script as it is, and create a new ticket for dealing with AWS resource name constraints

Reply:
I should mention that I tested this by running the script from the command line and passing in branch names that passed/failed both of the checks (regex/length)